File: Dart/3.longest_substring_without_repeating_characters.dart
void main() {
  print(lengthOfLongestSubstring("abcabcbb"));
  print(lengthOfLongestSubstring("bbbbb"));
  print(lengthOfLongestSubstring("pwwkew"));
}

int lengthOfLongestSubstring(String s) {
  var set = <String>{};

  for (int i = 0; i < s.length; i++) {
    final currentSet = <String>{};
    currentSet.add(s[i]);

    for (int j = i + 1; j < s.length; j++) {
      if (!currentSet.add(s[j])) {
        break;
      }
    }

    if (set.length < currentSet.length) {
      set = currentSet;
    }
  }

  return set.length;
}

File: src/3.longest_substring_without_repeating_characters.kt
fun main() {
    println(lengthOfLongestSubstring("abcabcbb"))
    println(lengthOfLongestSubstring("bbbbb"))
    println(lengthOfLongestSubstring("pwwkew"))
}

fun lengthOfLongestSubstring(s: String): Int {
    var set = mutableSetOf<String>()

    for (i in 0..<(s.length)) {
        val currentSet = mutableSetOf<String>()
        currentSet.add(s[i].toString())

        for (j in i + 1..<(s.length)) {
            if (!currentSet.add(s[j].toString())) {
                break
            }
        }

        if (set.size < currentSet.size) {
            set = currentSet
        }
    }

    return set.size
}